### DISCUSSION TITLE
v0.2.0 release

### DISCUSSION THREAD

**Discussion by skjelten** [skjelten/emusc#58] [General]
The current git master code is **way** better than the current v0.1.0 release and we should try to make another release as soon as possible. Before we make the release it makes sense to gather some feedback if things works OK across different systems. My list of current known regressions are:
* Crystal (capital instrument 99) has clipping issues due to erroneous filter (TVF) values
* ~~Jetplane (variation of instrument 126) makes EmuSC crash (seems to be pitch envelope related)~~
* MIDI notes are sometimes delayed, especially when playing MIDI songs with high number of simultaneous notes. This seems to stem from very high single core CPU usage, a long standing issue that was further made worse with the introduction of proper resampling.

Any other regressions that has been detected? And are other having CPU bound issues with complex MIDI-files?

**Comment by skjelten** [comment]
V.0.2.0 released. Feel free to use this thread for any discussions related to this release.

**Reply by shingo45endo** [reply]
Congratulations on the release of the new version. I have only recently been able to try this app, so I can't mention any differences from the previous version, but I have found some problems with the latest version and will report them later. Also, I have just prepared a build environment and will fix them and make PR if possible.

**Reply by shingo45endo** [reply]
I found a bug that causes a crash when playing Concert Cymbal (Note No. 59) in ORCHESTRA set. Since it is not reproduced in Ver. 0.1.0, the bug seems to have been introduced in the change to Ver. 0.2.0. 

Here's a SMF to reproduce: [ConcertCymCausesCrash.zip](https://github.com/skjelten/emusc/files/15443592/ConcertCymCausesCrash.zip)

Wave No. 243 "Con_sym" used in "Concert Cym." is the last wave of 244, and sample No. 992 used in "Con_sym" is also the last sample of 993. I have not yet looked at the implementation properly, but it might be due to a sort of off-by-one error.

**Reply by skjelten** [reply]
Thank you for the report, I can confirm that it crashes on my machine as well. I will have a look at it when time permits, but the last couple of weeks there's been a heat wave in my area, and emusc development is for me mostly a "winter activity" (northern hemisphere). Just FYI.

**Reply by skjelten** [reply]
I finally took some time to properly investigate this issue to figure out the root cause of the crash. Turns out that there are a few sample definitions, such as the one used for "Con_sym" partial, that have LoopLength > SampleLength. This caused a crash as a negative index was used to access the array with raw samples.

I am not sure how it could possibly make any sense to have LoopLength > SampleLength, but for now I will just add a test to make sure the index >= 0.

Thank you for making the midi-file with the crash report, that made it a lot easier to find the root cause. I also found another small bug caused by the SysEx message in your midi file that I will fix later.

**Comment by ThatRetroGuy** [comment]
hello! the version 2 release got some nice improving on the sound but stutter still seems to be a problem, but im sure some bug fixes will happen eventually.

**Reply by skjelten** [reply]
I would guess the stutter you hear, especially in v0.2.0, comes from CPU bound limitations. Hopefully, when we figure out how to use the lookup tables and implement more efficient code, this problem will go away.

**Reply by ThatRetroGuy** [reply]
well, if thats the situation,  hope that gets improved.
